torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7811,6 +7811,80 @@ def sample_inputs_diagonal_diag_embed(op_info, device, dtype, requires_grad, **k
     for shape, kwarg in chain(product(shapes_2d, kwargs_2d), product(shapes_3d, kwargs_3d)):
         yield SampleInput(make_arg(shape), kwargs=kwarg)
 
+def reference_inputs_diagonal(op_info, device, dtype, requires_grad, **kwargs):
+    yield from sample_inputs_diagonal_diag_embed(
+        op_info, device, dtype, requires_grad, **kwargs)
+
+    make_arg = partial(
+        make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    shapes2d = ((L, M),)
+    shapes3d = ((L, M, S),)
```

Comment:
empty inputs?

Reply:
already exists in `sample_inputs_diagonal_diag_embed`